CalcualtorLogic.js

```diff
@@ -135,17 +135,19 @@ class Calculator {
         this.resultElement.textContent = this.currentResult;
     }
     onOperatorButtonClick(event) {
-        // for selecting the clicked operator 
-        this.operator = event.target.dataset.operator;
-        // a flag helping in clearing the current result after clicking the next operands after operator buttons 
-        this.lastButtonClicked = "operator";
-        // assigning the current result which includes the first operand and the operator in new var for some purposes later
-        this.lastValue = this.currentResult;
+            // for selecting the clicked operator 
+            this.operator = event.target.dataset.operator;
+            // a flag helping in clearing the current result after clicking the next operands after operator buttons 
+            this.lastButtonClicked = "operator";
+            // assigning the current result which includes the first operand and the operator in new var for some purposes later
+            this.lastValue = this.currentResult;
 
-    }
-    onClickButtonClick(event) {
+        }
+        //onClearButtonClick() {
+    onClearButtonClick(event) {
         //this. = event.target.;
         this.lastValue = "";
+        this.currentResult = "";
```

Comment:
you need to update the UI also , which method you need to call here also?

Reply:
you also need to ensure that `onClearButtonClick` bound to the instance of the Calculator class 
how are going to do that based on what you have learnt so far??